### PR TITLE
Remove engine.Job from Start and Create

### PR DESCRIPTION
**Pull request by calavera**
Part of #12151

Signed-off-by: David Calavera david.calavera@gmail.com

### CODE REVIEW

**Comment by icecrime** [comment]
@calavera I think you only checked `test-unit` and `test-integration-cli`, unfortunately our old `test-integration` suite is still part of the campaign, and those are very broken :wink:

File: api/server/server.go [hunk 1]
```diff
@@ -790,30 +790,23 @@ func postContainersCreate(eng *engine.Engine, version version.Version, w http.Re
...
-		outWarnings  []string
-		stdoutBuffer = bytes.NewBuffer(nil)
-		warnings     = bytes.NewBuffer(nil)
+		warnings []string
+		name     = r.Form.Get("name")
+		env      = new(engine.Env)
 	)
 
-	if err := job.DecodeEnv(r.Body); err != nil {
+	if err := env.Decode(r.Body); err != nil {
```
Comment (LK4D4) [inline api/server/server.go]: Omg, can we replace it with types? Because it is even worse than jobs.

Comment (calavera) [inline api/server/server.go]: I agree that we should decode the body into a typed structure. I'd rather not doing it here to keep the changes to the minimum.

Comment (LK4D4) [inline api/server/server.go]: Then I propose to make it first.

Comment (calavera) [inline api/server/server.go]: I don't think making the env typed should block this PR. I think it should be a separated discussion. It touches more areas than just these two commands and we should be careful.

Comment (crosbymichael) [inline api/server/server.go]: I think it was unclear in the issue that removing jobs involved getting ride of everything in /engine including env.  Env is the worst part of the framework and actually the primary goal of the removal.

Comment (cpuguy83) [inline api/server/server.go]: Isn't the primary purpose of this to get rid of env in the first place

Comment (calavera) [inline api/server/server.go]: I understand that the purpose of all this is to get rid of everything under /engine. I feel like it's better to do it in phases, that's what I though was the idea/plan:

1. Get rid of all references to Job.
2. Replace env and dynamic lookups with type safe decoding.

I rather follow that script than this other one:

1. Get rid of some job references.
2. Replace env and dynamic lookups with type safe decoding.
3. Replace env references in the remaining jobs with the new type safe decoding.
4. Get rid of remaining jobs references.

Comment (LK4D4) [inline api/server/server.go]: @calavera I don't know where you seeing second script.
Current state of PR makes no sense. It makes nothing better, you replaced job with env, which will be deleted anyway and have same problems and even worse. It is possible to replace job with typed structs and your PR became smaller then now.

Comment (calavera) [inline api/server/server.go]: @LK4D4 I didn't replace anything, job calls env internally, I removed the layer on top of the env. It's not worse, we're removing the references to job, that's good.

Replacing the job with typed structs requires to go through all the project's code base to remove the reference to env before merging this. That's the second script.

**Comment by calavera** [comment]
OMG this is green!

@LK4D4 tons of changes to remove those pesky env references. Do you mind to double check and add some suggestions?

**Comment by LK4D4** [comment]
@calavera Thank you very much. I'll check soon.

**Comment by LK4D4** [comment]
LGTM

**Comment by cpuguy83** [comment]
LGTM

**Comment by calavera** [comment]
:metal:

**Pull request merged**